.github/workflows/doc-build.yml

```diff
@@ -1,7 +1,6 @@
-name: Build and Deploy Documentation
-
-on: [push, pull_request]
+name: Build Documentation
 
+on: [workflow_call, pull_request]
```

Comment:
why did you remove the `on push`?
when exactly does the `workflow_call` get triggered? I remember to read that people had problems with the `workflow_call` trigger

Reply:
It's removed here, because I added it to the `Publish Documentation` workflow.
So this way on a PR the documentation is built, but not published anywhere.
And only on a push to main the `Publish Documentation` workflow is called (which triggers the build by using the `Build Documentation` workflow).

Reply:
workflow_call is required for reusable workflows:
https://docs.github.com/en/actions/sharing-automations/reusing-workflows#creating-a-reusable-workflow